test/integration/crud/bulk.test.js

```diff
@@ -22,6 +27,86 @@ describe('Bulk', function () {
   before(function () {
     return setupDatabase(this.configuration);
   });
+  describe('BulkOperationBase', () => {
+    describe('#raw', function () {
+      let client;
+      beforeEach(async function () {
+        client = this.configuration.newClient();
+        await client.connect();
+      });
+      afterEach(async function () {
+        await client.close();
+      });
+      context('when called with an undefined operation', function () {
+        it('should throw a MongoInvalidArgument error ', async function () {
+          const bulkOp = client.db('test').collection('test').initializeUnorderedBulkOp();
+          expect(() => bulkOp.raw(undefined)).to.throw(MongoInvalidArgumentError);
+          expect(() => bulkOp.raw(true)).to.throw(MongoInvalidArgumentError);
+          expect(() => bulkOp.raw(3)).to.throw(MongoInvalidArgumentError);
+        });
+
+        it('should throw an error with the specifc message: "Operation must be an object with an operation key"', async function () {
+          const bulkOp = client.db('test').collection('test').initializeUnorderedBulkOp();
+          try {
+            bulkOp.raw(undefined);
+            expect.fail(
+              'Expected passing argument of "undefined" to .raw to throw error, failed to throw error'
+            );
+          } catch (error) {
+            expect(error.message).to.equal('Operation must be an object with an operation key');
+          }
+        });
+      });
+
+      context('when called with a valid operation', function () {
+        it('should not throw a MongoInvalidArgument error', async function () {
+          try {
+            client.db('test').collection('test').initializeUnorderedBulkOp().raw({ insertOne: {} });
+          } catch (error) {
+            expect(error).not.to.exist;
+          }
+        });
+      });
+    });
+  });
+
+  describe('Db.collection', function () {
+    describe('#insertMany', function () {
+      let client;
+      beforeEach(async function () {
+        client = this.configuration.newClient();
+        await client.connect();
+      });
+      afterEach(async function () {
+        await client.close();
+      });
+      context('when passed an invalid docs argument', function () {
+        it('insertMany should throw a MongoInvalidArgument error when called with a invalid operation', async function () {
+          try {
+            const docs = [];
+            docs[1] = { color: 'red' };
+            await client.db('test').collection('test').insertMany(docs);
+            expect.fail('Expected insertMany to throw error, failed to throw error');
+          } catch (error) {
+            expect(error).to.be.instanceOf(MongoInvalidArgumentError);
```

Comment:
I think we should also check the error message here.  Since the reported bug relates to the error message thrown, we should confirm here that we're now throwing the better error message 😄

Reply:
done